src/main/java/redis/clients/jedis/JedisShardInfo.java

```diff
@@ -91,9 +102,21 @@ public void setTimeout(int timeout) {
   public String getName() {
     return name;
   }
+  
+  public int getDb() {
+    return db;
+  }
 
   @Override
   public Jedis createResource() {
-    return new Jedis(this);
+    Jedis jedis = new Jedis(this);
```

Comment:
I agree with @HeartSaVioR solution. It's more elegant to set the DB at BinaryJedis level where initialization takes place rather to make it elsewhere. @nykolaslima please go ahead and make the changes and let us know once we can review.